token/cli/src/main.rs

```diff
@@ -550,10 +571,11 @@ async fn command_authorize(
         AuthorityType::WithheldWithdraw => "withdraw withheld authority",
         AuthorityType::InterestRate => "interest rate authority",
     };
-    let (previous_authority, program_id) = if !config.sign_only {
+
+    let previous_authority = if !config.sign_only {
         let target_account = config.rpc_client.get_account(&account).await?;
         config.check_owner(&account, &target_account.owner)?;
-        let program_id = target_account.owner;
```

Comment:
Be sure to keep this and update the token client based on the owner, since the default is just `Tokenkeg...`

Reply:
ty i didnt even think of this, id gotten used to "providing the correct program id on invocation is mandatory"

Reply:
No worries, it's really not clear in the code... we may want to put a note on `config.program_id` to discourage its use as much as possible